pkgs/development/coq-modules/coq-lsp/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, mkCoqDerivation, coq, serapi, version ? null }:
+
+mkCoqDerivation rec {
+  pname = "coq-lsp";
+  owner = "ejgallego";
+
+  useDune = true;
+
+  release."0.1.4+v8.16".sha256 = "sha256-5jztvAKpCFehtxu9Kw2inbPUZYVFIMcIoD5/u1WBuyI=";
+
+  inherit version;
+  defaultVersion = with lib.versions; lib.switch coq.coq-version [
+    { case = range "8.16" "8.17"; out = "0.1.4+v8.16"; }
+  ] null;
```

Comment:
Folks thanks for this PR, for Coq 8.17 you need a 8.17 specific tree, unfortunately the Coq API we use is not stable between versions.

Tag for 8.17 is 0.1.4+v8.17

Reply:
Then it should be possible to use `"0.1.4+v${coq.version}"`, that's the beauty about a functional language

Reply:
It should yes! Something that has been brought to my attention is whether the branch names should be `+v8.16` or `+8.16`, it seems the lastest scheme is the standard (including in some of my packages) so I'm unsure if I should tweak the tag names for the upcoming releases (and add the corresponding aliases back)

Any advice you folks have here as to easy packaging is most welcome.

Reply:
It would be preferable, as I think the v messes up the reverse package name parser (e.g. when doing `nix-env -e coq-lsp` it tries to strip the version suffix)

Reply:
Thanks @Thesola10 , so to confirm `0.1.4+8.16` is Nix-friendly ?

Reply:
You might not need to change your actual tagging scheme -- since `mkDerivation` accepts arbitrary key/values, you could compose the derivation's `version` as `"${packageVersion}+${coq.coq-version}"` and the source tag as `"${packageVersion}+v${coq.coq-version}"`